sources/platform/integrations/programming/api.md

```diff
@@ -108,24 +106,55 @@ Other dependencies are more complicated, and **it is your responsibility that th
-
 :::
 
-### Actor execution and scheduling
+### Actor execution
+
+When you run an Actor, Apify creates a new, short-lived run API token, and injects it into the Actor environment. This applies to scoped tokens as well, so when you run an Actor with a scoped token, **the Actor is executed with a different token with a different scope.**
+
+In the scoped token configuration you can choose what scope the run API token gets, effectively determining what the Actor can access during its run.
+
+Apify currently supports two modes:
+
+- **Full access**: Allow Actors to access all your account's data.
+- **Restricted access**: Restrict what Actors can access using the scope of this Actor.
 
-When you run an Actor with a scoped token (or schedule one), Apify will inject a new, _unscoped_ token to the Actor. This means that **the Actor will have full access to all resources in your account**.
+![Choose permission mode for running Actors with a scoped token](../images/api-token-scoped-run-modes.png)
 
-This way you can be sure that once you give a token the permission to run an Actor, it will just work, and you don't have to worry
-about the exact permissions the Actor might need. However, this also means that you need to trust the Actor.
+#### Full access: Allow Actors to access all your account's data
 
-:::note
+When you run an Actor with a scoped token in this mode, Apify will inject an _unscoped_ token to the Actor runtime.
 
-This is why **we currently do not allow scoped tokens to create or modify Actors**. With those permissions it would be easy for the token to upload malicious code and gain access to your full account. If you do need to create or modify Actors via Apify API, you need to use an unscoped token.
+This way you can be sure that once you give a token the permission to run an Actor, it will just work, and you don't have to worry about the exact permissions the Actor might need. However, this also means that you need to trust the Actor.
 
+:::tip
+Use this mode if you want to integrate with a 3rd-party service to run your Actors. Create a scoped token that can only run the Actor you need, and share it with the service. Even if the token is leaked, it can't be used to access your other data.
+:::
+
+#### Restricted access: Restrict what Actors can access using the scope of this Actor
+
+When you run an Actor with a scoped token in this mode, Apify will inject a token with the same scope as the scope of the original token.
+
+This way you can be sure that Actors won't accidentally—or intentionally—access any data they shouldn't. However, Actors might not function properly if the scope is not sufficient.
+
+:::tip
+This restriction is _transitive_, which means that if the Actor runs another Actor, its access will be restricted as well.
 :::
 
+#### Default run storages
+
+When Apify runs an Actor, it automatically creates a set of default storages (a dataset, a key-value store and request queue) that the Actor can use in runtime.
```

Comment:
Maybe link here to each part of documentation?

Reply:
Good idea, added.

Reply:
Oh I thought more to dataset, request queue and Key-value store ( or maybe just one link under storages?)

Reply:
Fair enough, it works either way